kernel/src/hil/symmetric_encryption.rs

```diff
@@ -0,0 +1,16 @@
+use returncode::ReturnCode;
+
+pub trait SymmetricEncryptionDriver {
+    fn init(&self);
+    fn set_key(&self, key: &'static mut [u8]);
+    // fn encrypt(&self, plaintext: &'static mut [u8], len: u8);
+    // fn decrypt(&self, ciphertext: &'static mut [u8], len: u8);
+    fn crypt_ctr(&self, data: &'static mut [u8], init_ctr: &'static mut [u8], len: u8);
```

Comment:
It is used both for encryption and decryption, since the only the encryption mode of AES is used i.e. I use a statusflag to indicate enc/dec instead.